src/DotNetWorker.Core/FunctionMetadata/IFunctionMetadata.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections.Generic;
+
+namespace Microsoft.Azure.Functions.Worker.Core.FunctionMetadata
+{
+    /// <summary>
+    /// Public interface that represents properties in function metadata.
+    /// </summary>
+    public interface IFunctionMetadata
+    {
+        /// <summary>
+        /// Unique function id.
+        /// </summary>
+        string FunctionId { get; set; }
+
+        /// <summary>
+        /// If function is a proxy function return true, else return false.
```

Comment:
nit - not a big deal, but could make the comments the same for the interface & implementation classes?